admin/docs/tailwindcss.md

```diff
@@ -13,10 +13,27 @@ components, you can do so by running Tailwind's build process in your host
 application.
 
 This process presumes that you have a working knowledge of Tailwind CSS. If you
-are not familiar with Tailwind, please refer to the [Tailwind
-documentation](https://tailwindcss.com/docs) for more information.
+are not familiar with Tailwind, please refer to the
+[Tailwind documentation](https://tailwindcss.com/docs) for more information.
 
-## Setting up a local TailwindCSS build for Solidus Admin
+## Coding Style
+
+The provided Tailwind CSS configuration is as vanilla as possible, with little to no customizations.
+In the same way we try to stay within the boundaries of the default Tailwind CSS classes and avoid custom classes,
+occasionally we need to add custom styles and in that case we try to use dynamic square brackets classes.
+
+CSS classes are all kept in the HTML templates of the components in order to simplify the configuration.
+
+The few customizations that we have are added through plugins since in addition to classes they support variants.
+
+## Releasing with a precompiled Tailwind CSS
+
+The Tailwind CSS stylesheets are precompiled and included in the gem.
+This means that the gem can be used without having to compile Tailwind CSS in the host application.
+The compiled file is not included in the git repository, but it is generated and included in the gem as
```

Comment:
What do you think about committing the built tailwind css file into the admin repo (and update it on PR merge)? That file won't change much (at least once the admin has been built out). And even in until then it makes to have it updated regularly (not just on gem release).

People might use solidus_admin from git source like this

```
# Gemfile
gem "solidus_admin", github: "solidusio/solidus", branch: "v4.3"
```

in order to get bug fixes early for stable versions that have not been released yet.

We want to make sure they get the admin css file, right?

Reply:
I have a GH workflow that builds a JS bundle for Alchemy CMS on every merge (and when the source files have changed) https://github.com/AlchemyCMS/alchemy_cms/blob/main/.github/workflows/build.yml

I am, happy to contribute this to solidus if we agree on adding the file into the repo

Reply:
@tvdeyen I really didn't want to add such a noisy file, but I admin it might be the easiest thing to do. Do you usually update it manually in Alchemy and then use the GH action as a fallback or you rely on it as the main thing that updates the file?

Reply:
> I really didn't want to add such a noisy file

I won't consider this file noise (at least until the new admin is done). What is the difference to other assets we have committed into our repo?

> Do you usually update it manually in Alchemy and then use the GH action as a fallback or you rely on it as the main thing that updates the file?

I manually update the bundle whenever I add a dependency. The GH action is used for automated dependency updates (like dependabot or depfu)

Reply:
Today we discussed this topic during the core team meeting, and the consensus seems to be to try to avoid directly committing the generated CSS artifact file to this repository. One interesting option that came up is replacing tailwindCSS with https://unocss.dev/, which should remove the issue. I'll try to investigate more on this.

Reply:
@tvdeyen I'm thinking that regardless of the solution we'll be able to find for the sandbox issue, this PR should be progressed since it includes documentation on many other topics and, speaking of the problematic new admin `tailwind.css`, it documents the current behavior, which otherwise would remain rather unexposed. We have https://github.com/solidusio/solidus/issues/5598 for tracking the sandbox issue and exploring its solutions.